control-plane/pkg/reconciler/brokeryolo/broker.go

```diff
@@ -0,0 +1,294 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package brokeryolo
+
+import (
+	"context"
+	"fmt"
+	mf "github.com/manifestival/manifestival"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/utils/pointer"
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/config"
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/kafka"
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/prober"
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/base"
+	brokerreconciler "knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/broker"
+	eventing "knative.dev/eventing/pkg/apis/eventing/v1"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/reconciler"
+	"knative.dev/pkg/resolver"
+)
+
+// TODO: 1.
+// TODO: Read config-kafka-broker-data-plane in user namespace first, fallback to the one in knative-eventing.
+// TODO: Though, this would mean that we need to mount the configmap from knative-eventing to the pod in user namespace.
+// TODO: This is not supported by Kubernetes. If we want this, we need to read the configmap content dynamically, with
+// TODO: necessary RBAC.
+
+// TODO: 2.
+// TODO: config-kafka-broker-data-plane is created by the reconciler dynamically.
+// TODO: This means, the contents can ONLY be changed by users after a broker is created.
+// TODO: Also, reconciler will reconcile it when it is changed.
+
+// TODO: 3.
+// TODO: config-tracing and config-logging has the same mounting issues. We need to read these manually.
+
+// TODO: 4.
+// TODO: Do we reconcile everything when one of the configmaps change?
+// TODO: Specifically, contract configmap, etc.
+
+type Reconciler struct {
+	*base.Reconciler
+	*config.Env
+
+	Resolver *resolver.URIResolver
+
+	ConfigMapLister corelisters.ConfigMapLister
+
+	// NewKafkaClusterAdminClient creates new sarama ClusterAdmin. It's convenient to add this as Reconciler field so that we can
+	// mock the function used during the reconciliation loop.
+	NewKafkaClusterAdminClient kafka.NewClusterAdminClientFunc
+
+	BootstrapServers string
+
+	// BaseDataPlaneManifest is a Manifestival manifest that has the resources that need to be created in the user namespace.
+	BaseDataPlaneManifest mf.Manifest
+
+	Prober prober.Prober
+
+	IPsLister prober.IPListerWithMapping
```

Comment:
why not `map[Namespace: string]prober.Prober` instead of 2 coupled objects

Reply:
I don't want to maintain the list of probers per namespaces that I might possibly stop/start.

Maintaining the list of ingress addresses of the brokers and using the same prober makes more sense to me.